site/en/docs/extensions/mv3/content_scripts/index.md

```diff
@@ -200,66 +205,78 @@ the page it's trying to inject scripts into. Host permissions can either be gran
 requesting them as part of your extension's manifest (see [`host_permissions`][33]) or temporarily
 via [activeTab][15].
 
-Below we'll look at an example that uses activeTab.
+Below we'll look at a couple different versions of an activeTab-based extension.
 
-```json/3-5
+```json/4-6
+//// manifest.json ////
 {
   "name": "My extension",
   ...
   "permissions": [
     "activeTab"
   ],
-  ...
+  "background": {
+    "service_worker": "background.js"
+  }
 }
 ```
 
-Content scripts can be injected as files.
+Content scripts can be injected as files…
 
 ```js
-chrome.runtime.onMessage.addListener((message, callback) => {
-  if (message == "runContentScript"){
-    chrome.scripting.executeScript({
-      file: 'contentScript.js'
-    });
-  }
+//// content-script.js ////
+document.body.style.backgroundColor = 'orange';
+```
+
+```js
+//// background.js ////
+chrome.action.onClicked.addListener((tab) => {
+  chrome.scripting.executeScript({
+    target: { tabId: tabId },
+    files: ['content-script.js']
+  });
 });
 ```
 
-Or an entire file can be injected.
+…or a function body can be injected and executed as a content script.
 
 ```js
+//// background.js ////
 function injectedFunction() {
   document.body.style.backgroundColor = 'orange';
 }
 
-chrome.runtime.onMessage.addListener((message, callback) => {
-  if (message == "changeColor"){
-    chrome.scripting.executeScript({
-      function: injectedFunction
-    });
-  }
+chrome.action.onClicked.addListener((message, callback) => {
+  chrome.scripting.executeScript({
+    target: { tabId: tabId },
+    function: injectedFunction
+  });
 });
 ```
 
+Be aware that the injected function is a copy the function referenced in the
+`chrome.scripting.executeScript` call, not the original function itself. As a result, the function's
+body must be self contained; references to variables outside of the function will cause the content script to throw a [`ReferenceError`][ref-error].
+
+{% if false %}
```

Comment:
This part looks pretty sane and understandable. Why remove it?

Reply:
This feature is not yet implemented.